client/reader/search-stream/site-results.jsx

```diff
@@ -81,14 +81,42 @@ export default connect(
 			sort: ownProps.sort,
 		} );
 
-		// Check if searchResults has one item and if it has a feed_ID
-		if ( searchResults && searchResults.length === 1 ) {
-			let feed = searchResults[ 0 ];
-			if ( feed.feed_ID.length > 0 ) {
-				// If it has a feed_id, get the feed object from the state
-				feed = getFeed( state, feed.feed_ID );
-			}
-			ownProps.onReceiveSearchResults( feed );
+		// Check if searchResults has any feeds
+		if ( searchResults && searchResults.length > 0 ) {
+			const feeds = searchResults;
+			// We want to create a list of unique feeds based on searchResults
+			// We need to do this because the search results may contain duplicate feeds URLs with different http or https schemes
+			// eslint-disable-next-line wpcalypso/redux-no-bound-selectors
+			const feedResults = feeds.reduce( ( uniqueFeeds, feed ) => {
```

Comment:
hmm I wonder if there are other areas where this will be a problem? It looks like the "manage following" page also uses the feed search, as does the edit page for custom lists. I think we may need to move this logic either to the back end or somewhere it can be applied for all of these cases